packages/waggle/src/storage/storage.ts

```diff
@@ -344,49 +310,85 @@ export class Storage {
   }
 
   public async subscribeForChannel(
-    channelAddress: string,
-    channelInfo?: IChannelInfo
+    channel: IChannelInfo
   ): Promise<void> {
     let db: EventStore<IMessage>
-    let repo = this.publicChannelsRepos.get(channelAddress)
-
+    let repo = this.publicChannelsRepos.get(channel.address)
     if (repo) {
       db = repo.db
     } else {
-      db = await this.createChannel(channelAddress, channelInfo)
+      db = await this.createChannel(channel)
       if (!db) {
-        log(`Can't subscribe to channel ${channelAddress}`)
+        log(`Can't subscribe to channel ${channel.address}`)
         return
       }
-      repo = this.publicChannelsRepos.get(channelAddress)
+      repo = this.publicChannelsRepos.get(channel.address)
     }
 
     if (repo && !repo.eventsAttached) {
-      log('Subscribing to channel ', channelAddress)
+      log('Subscribing to channel ', channel.address)
+
       db.events.on('write', (_address, entry) => {
-        log(`Writing to public channel db ${channelAddress}`)
+        log(`Writing to public channel db ${channel.address}`)
         socketMessage(this.io, {
           message: entry.payload.value,
-          channelAddress: channelAddress,
+          channelAddress: channel.address,
           communityId: this.communityId
         })
       })
+
       db.events.on('replicate.progress', (_address, _hash, entry, _progress, _total) => {
         log('Message replicated')
         socketMessage(this.io, {
           message: entry.payload.value,
-          channelAddress: channelAddress,
+          channelAddress: channel.address,
           communityId: this.communityId
         })
       })
+
       db.events.on('ready', () => {
         const ids = this.getAllEventLogEntries(db).map(msg => msg.id)
-        sendIdsToZbay(this.io, { ids, channelAddress, communityId: this.communityId })
+        sendIdsToZbay(this.io, { ids, channelAddress: channel.address, communityId: this.communityId })
       })
+
       repo.eventsAttached = true
       const ids = this.getAllEventLogEntries(db).map(msg => msg.id)
-      sendIdsToZbay(this.io, { ids, channelAddress, communityId: this.communityId })
+      sendIdsToZbay(this.io, { ids, channelAddress: channel.address, communityId: this.communityId })
```

Comment:
do we need tests at the waggle level to make sure created channels are replicated?

Reply:
I believe Bartek added those tests a long time ago (?). I'll ask him

Reply:
okay!

Reply:
We think it's not us who should take care of the replication tests. It's up to the orbitdb folks